File: 2351-first-letter-to-appear-twice/2351-first-letter-to-appear-twice.js
/**
 * @param {string} s
 * @return {character}
 */
var repeatedCharacter = function(s) {
     const mp= new Set();
    for(i of s)
        {
            if(mp.has(i))
                return i;
            
            mp.add(i);
        }
    return '\0';
};

File: 2351-first-letter-to-appear-twice/2351-first-letter-to-appear-twice.cpp
class Solution {
public:
    char repeatedCharacter(string s) {
        map<char,int> mp;
        char ans;
     
        for(auto x: s)
        {
            mp[x]++;
            if(mp[x]>1)
            {
               ans= x;
                break;
            }
        }
        return ans;
    }
};